uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -545,13 +625,166 @@ define([
       /**
        * checks if all params are set to calculate balance
        *
-       * @param {Array} listOfDayTypes collection of available day types
+       * @param {Boolean} true if all present else false
        */
       function canCalculateChange() {
         return vm.leaveRequest.from_date && vm.leaveRequest.to_date &&
           vm.leaveRequest.from_date_type && vm.leaveRequest.to_date_type;
       }
 
+      /**
+       * checks if leaverequest is managed by given manager id
+       *
+       * @param {String} managerContactId
+       * @returns {Promise}
+       */
+      function isManagedBy(managerContactId) {
+        return vm.leaveRequest.roleOf({
+            id: managerContactId
+          })
+          .then(function (role) {
+            if (role === 'manager') {
+              vm.uiOptions.isManager = true;
+            } else {
+              vm.uiOptions.isManager = false;
+            }
+          });
+      }
+
+      /**
+       * initialize absence types
+       *
+       */
+      function initAbsenceType() {
+        if (canEdit()) {
+          vm.selectedAbsenceType = _.find(vm.absenceTypes, function (absenceType) {
+            return absenceType.id == vm.leaveRequest.type_id;
+          });
+        } else {
+          // Assign the first absence type to the leave request
+          vm.selectedAbsenceType = vm.absenceTypes[0];
+        }
+
+        vm.leaveRequest.type_id = vm.selectedAbsenceType.id;
```

Comment:
Shouldn't this happen only in the `else` branch? If the leave request already exists, why are you reassigning the `type_id`?